auth/actions.ts

```diff
@@ -0,0 +1,39 @@
+'use server';
+
+import {
+  KEY_CALLBACK_URL,
+  KEY_CREDENTIALS_SIGN_IN_ERROR,
+  KEY_CREDENTIALS_SIGN_IN_ERROR_URL,
+  signIn,
+  signOut,
+} from '@/auth';
+import { redirect } from 'next/navigation';
+
+export const signInAction = async (
+  _prevState: string | undefined,
+  formData: FormData
+) => {
+  try {
+    await signIn('credentials', Object.fromEntries(formData));
+  } catch (error) {
+    if (
+      `${error}`.includes(KEY_CREDENTIALS_SIGN_IN_ERROR) ||
+      `${error}`.includes(KEY_CREDENTIALS_SIGN_IN_ERROR_URL)
+    ) {
+      // Return credentials error to display on sign-in page.
+      return KEY_CREDENTIALS_SIGN_IN_ERROR;
+    } else if (!`${error}`.includes('NEXT_REDIRECT')) {
```

Comment:
This `else if` doesn't return. You mentioned re-throwing non-redirect errors. This will still hit the throw below, right?